src/lib/provable/gadgets/sha256.ts

```diff
@@ -88,80 +88,22 @@ const SHA256 = {
     // padding the message $5.1.1 into blocks that are a multiple of 512
     let messageBlocks = padding(data);
 
-    const H = SHA256Constants.H.map((x) => UInt32.from(x));
+    let H = SHA256Constants.H.map((x) => UInt32.from(x));
```

Comment:
Since these are constants, we should keep them that - you can create a second variable further down that takes the return value from `computeMessageScheduele`

Reply:
Before I push again, what do you think about the changes below

```ts
 const SHA256 = {
  hash(data: FlexibleBytes) {
    // preprocessing §6.2
    // padding the message $5.1.1 into blocks that are a multiple of 512
    let messageBlocks = padding(data);

    let H = SHA256.initialState;
    const N = messageBlocks.length;

    for (let i = 0; i < N; i++) {
      const W = createMessageSchedule(messageBlocks[i]);
      H = sha256Compression(H, W);
    }

    // the working variables H[i] are 32bit, however we want to decompose them into bytes to be more compatible
    // wordToBytes expects little endian, so we reverse the bytes
    return Bytes.from(H.map((x) => wordToBytes(x.value, 4).reverse()).flat());
  },
  compression: sha256Compression,
  createMessageSchedule,
  initialState: SHA256Constants.H.map((x) => UInt32.from(x)),
};
```

Reply:
That works for me!

Reply:
Exported `initialState` as a getter instead

```ts
get initialState() {
    return SHA256Constants.H.map((x) => UInt32.from(x));
  },
 ```